### PR TITLE
Build: Fix minor error-prone warnings

### PR DESCRIPTION
**Pull request by ajantha-bhat**
I have observed that the build [`./gradlew clean build -x test`] has some warnings. 
So it is an effort to keep the build green. 

Before:
Screenshot 2023-01-20 at 7 50 45 AM

After:
Screenshot 2023-01-20 at 8 04 01 AM

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
cc: @nastra, @Fokko

File: core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java [hunk 1]
```diff
@@ -329,14 +329,14 @@ static Long expiresAtMillis(String token) {
       return null;
     }
 
-    String[] parts = token.split("\\.");
-    if (parts.length != 3) {
+    List<String> parts = Splitter.on('.').splitToList(token);
```
Comment (nastra) [inline core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java:332]: nit: should we extract this to a `private static final Splitter DOT_SPLITTER`?

**Review by nastra — APPROVED** [review]
LGTM, thanks for fixing @ajantha-bhat

File: core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java (hunk 1 above)
Comment (ajantha-bhat) [reply core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java:332]: As it is not used in multiple places (in the same file), I thought no need for an extra variable.

**Comment by XN137** [comment]
should we raise the severity of the `StringSplitter` check to `ERROR` like [here](https://github.com/apache/iceberg/blob/4491e7d6d6911ca0ecf129d7528dda2717f8c938/baseline.gradle#L70) ?
this prevents similar problems from being introduced again

**Comment by ajantha-bhat** [comment]
> should we raise the severity of the StringSplitter check to ERROR like [here](https://github.com/apache/iceberg/blob/4491e7d6d6911ca0ecf129d7528dda2717f8c938/baseline.gradle#L70) ?
this prevents similar problems from being introduced again

We could add this. But I am not sure on what basis we add here because there are 400 plus bug patterns in errorProne. Even if we prevent this error, there can be some other error-prone warnings that can still prevent the build from looking green.

**Comment by XN137** [comment]
if these warnings are worth fixing, shouldnt they also be worth preventing?
automatic prevention is usually cheaper than manually observing and fixing future warnings.

this also has the added benefit that we are sure this PR addressed all these types of violations, not just the ones you happened to see.

i dont see how the existence of other errorprone checks, that are not being enforced, matters in this discussion.
ideally all of the checks we agree are worth fixing should be enforced... the problem is identifying those and addressing all their current violations.

**Review by Fokko — APPROVED** [review]
Great work @ajantha-bhat. I also did quite some of these fixes because they clutter up your output when building locally and in the CI.

I agree with @XN137 that we should find some way to get rid of the warnings by either turning them into errors or if they aren't important, just suppressing them.

**Comment by XN137** [comment]
> or if they aren't important, just suppressing them.

its probably what you mean, but instead of suppressing them (=adding annotations), if we dont agree with a check, it should be turned `OFF` instead, so it does not add to the compile time and does not add future warnings

**Comment by ajantha-bhat** [comment]
@Fokko, @XN137, @nastra: Thanks for the review. 
I have enforced `StringSplitter` cases to throw the error as discussed above and the builds are passing.

**Comment by Fokko** [comment]
@XN137 I agree with you, but there are some tricky cases where you want to use keep the check in place, but just suppress them at certain places. I think this should be checked on a per-rule basis.

**Pull request merged**